scripts/generate_variable_lists.py

```diff
@@ -0,0 +1,413 @@
+
+    Parameters
+    ----------
+    experiment_dict: dict
+        The dictionary containing all experiments and their associated variables.
+    experiment: str
+        The experiment whose variables are being updated.
+    variable_dict: dict
+        A dictionary to populate with the updated variable data.
+
+    Returns
+    -------
+    dict[str, str]
+        A dictionary of variable name and priority level key-value pairs for a single experiment.
+    """
+    priority_comments = set_priority_comments(experiment_dict, experiment)
+    all_labels = get_all_variables(experiment_dict, experiment)
+
+    # Check all labels against their priority status and label accordingly.
+    for var in all_labels:
+        comment = priority_comments.get(var, "")
+        variable_dict[var] = comment
+
+    return variable_dict
+
+
+def extract_variable_from_title(map: dict) -> str:
+    """Returns the variable name from the given title within the mapping dictionary for a single variable.
+
+    Parameters
+    ----------
+    map: dict
+        The mapping dictionary for a single variable.
+
+    Returns
+    -------
+    str
+        The variable name.
+    """
+    title = extract_mapping_information(map)["title"]
+
+    return re.search(r"Variable\s+([^\s(]+)", title).group(1)
+
+
+def identify_not_produced(mappings_dict: list[dict], variable_dict: dict[str, str]) -> dict[str, str]:
+    """Identify all variables marked as "do not produce" in a single experiment.
+
+    Parameters
+    ----------
+    mappings_dict: list[dict]
+        The dictionary containing mapping information for all variables.
+    variable_dict: dict[str, str]
+        The dictionary of name and priority level key-value pairs for a single experiment.
+
+    Returns
+    -------
+    dict[str, str]
+        An updated dictionary containing production status for variables marked "do-not-produce".
+    """
+    # Loop over all variables to find those labelled as "do-not-produce" and mark them as such.
+    for map in mappings_dict:
+        labels = extract_mapping_information(map)["labels"]
+        var = extract_variable_from_title(map)
+
+        # Overriding any existing "priority" value in the dict is acceptable since do-not-produce takes precedence.
+        if "do-not-produce" in labels:
+            variable_dict[var] = " # do-not-produce"
+
+    return variable_dict
+
+
+def get_streams(mappings_dict: list[dict]) -> dict[str, str]:
+    """Creates a dictionary for variables and their associated output stream for a single experiment.
+
+    Parameters
+    ----------
+    mappings_dict: list[dict]
+        The dictionary containing mapping information for all variables.
+
+    Returns
+    -------
+    dict[str, str]
+        A dictionary containing variables and their associated output stream.
+    """
+    streams = {}
+
+    # Access stash entries for each variable and check if it contains values.
+    for map in mappings_dict:
```

Comment:
be cautious about reusing built-in function names as variables -- this can lead to  confusion. 

I suggest `map` -> `mapping` and `var` to `variable` below to avoid this